std/typecons.d

```diff
@@ -3894,7 +3998,7 @@ if (Targets.length >= 1 && allSatisfy!(isMutable, Targets))
     {
         auto wrap(inout Source src)
         {
-            static assert(hasRequireMethods!(),
+            static assert(implementsInterface!(Source, Targets),
                           "Source "~Source.stringof~
                           " does not have structural conformance to "~
                           Targets.stringof);
```

Comment:
Hmmm, some more information here would be PURE GOLD. You may leave that for later, let us know.